pyo3-macros-backend/src/pyclass.rs

```diff
@@ -442,24 +478,105 @@ impl<'a> PyClassEnum<'a> {
     }
 }
 
-pub fn build_py_enum(
-    enum_: &mut syn::ItemEnum,
-    mut args: PyClassArgs,
-    method_type: PyClassMethodsType,
-) -> syn::Result<TokenStream> {
-    args.options.take_pyo3_options(&mut enum_.attrs)?;
+struct PyClassUnitEnumVariant<'a> {
+    ident: &'a syn::Ident,
+    options: EnumVariantPyO3Options,
+}
 
-    if let Some(extends) = &args.options.extends {
-        bail_spanned!(extends.span() => "enums can't extend from other classes");
-    } else if let Some(subclass) = &args.options.subclass {
-        bail_spanned!(subclass.span() => "enums can't be inherited by other classes");
-    } else if enum_.variants.is_empty() {
-        bail_spanned!(enum_.brace_token.span.join() => "#[pyclass] can't be used on enums without any variants");
+impl<'a> PyClassUnitEnumVariant<'a> {
+    fn python_name(&self, args: &PyClassArgs) -> Cow<'_, syn::Ident> {
+        self.options
+            .name
+            .as_ref()
+            .map(|name_attr| Cow::Borrowed(&name_attr.value.0))
+            .unwrap_or_else(|| {
+                let name = self.ident.unraw();
+                if let Some(attr) = &args.options.rename_all {
+                    let new_name = apply_renaming_rule(attr.value.rule, &name.to_string());
+                    Cow::Owned(Ident::new(&new_name, Span::call_site()))
+                } else {
+                    Cow::Owned(name)
+                }
+            })
```

Comment:
There's a little duplication here with the `python_name` for the complex variant, perhaps can be unified?